### ISSUE TITLE
Official Sentry Nest.js support in `8.x` of the Sentry SDK

### ISSUE THREAD

**Issue by AbhiPrasad** [ntegral/nestjs-sentry#123]
In `8.x` of the Sentry SDK, we added official Nest.js support: https://docs.sentry.io/platforms/javascript/guides/nestjs/

If you have any suggestions for features, please reach out!

**Comment by Simolation** [comment]
@AbhiPrasad Does it also support GraphQL in NestJS?

I tried setting it up using the Sentry.graphqlIntegration() integration, but the tracked data is entirely different from before and also does not really track GraphQL sessions. It just shows the exceptions which are thrown, for example due to missing Authentication.

**Comment by AbhiPrasad** [comment]
> but the tracked data is entirely different from before and also does not really track GraphQL sessions

Are you referring to errors or performance monitoring?

What specifically do you mean by tracking GraphQL sessions?

In general this seems like something we need to fix! If you open a GH issue on https://github.com/getsentry/sentry-javascript with results that you expect, we can address your use case and update the docs accordingly.

**Comment by Koslun** [comment]
> In `8.x` of the Sentry SDK, we added official Nest.js support: https://docs.sentry.io/platforms/javascript/guides/nestjs/
> 
> If you have any suggestions for features, please reach out!

The primary point holding our org back is ambiguity in enriching events, the [current docs](https://docs.sentry.io/platforms/javascript/guides/nestjs/enriching-events/transaction-name/) still refer to deprecated API calls and from the docs it isn't clear there's any help in gathering details that we're gathering now. Including transactions with route paths, status codes and tagging of services/controller affected. A switch is however interesting as a lot of this information is working but not well with inexplicably incorrect information for certain code paths.

Feedback & feature requests best submitted as GH issues as well?

If things are resolved by Q3 it might be worth for us to step and contribute.

**Comment by AbhiPrasad** [comment]
> Feedback & feature requests best submitted as GH issues as well?

Yes please - we'll get an SDK engineer to help tackle your request.

> The primary point holding our org back is ambiguity in enriching events, the [current docs](https://docs.sentry.io/platforms/javascript/guides/nestjs/enriching-events/transaction-name/) still refer to deprecated API calls and from the docs it isn't clear there's any help in gathering details that we're gathering now. 

We were planning to deprecate this API in `7.x`, but decided against it, so we reverted that decision. The latest release of `7.x` should not have `setTransactionName` as deprecated API. If there are any examples of unclear docs, please let us know, we will fix ASAP! (PRs are also welcome, [docs are open source](https://github.com/getsentry/sentry-docs)).

>Including transactions with route paths, status codes and tagging of services/controller affected. A switch is however interesting as a lot of this information is working but not well with inexplicably incorrect information for certain code paths.

our integration is new, so we do have some improvements to data quality to make here. Any and all feedback very much appreciated.
